torchmetrics/image/ms_ssim.py

```diff
@@ -0,0 +1,151 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, List, Optional, Sequence, Tuple, Union
+
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.image.ms_ssim import _ms_ssim_compute
+from torchmetrics.functional.image.ssim import _ssim_update
+from torchmetrics.metric import Metric
+from torchmetrics.utilities import rank_zero_warn
+from torchmetrics.utilities.data import dim_zero_cat
+
+
+class MultiScaleSSIM(Metric):
+    """Computes `MultiScaleSSIM`_, Multi-scale Structual Similarity Index Measure, which is a generalization of
+    Structual Similarity Index Measure by incorporating image details at different resolution scores.
+
+    Args:
+        kernel_size: size of the gaussian kernel
+        sigma: Standard deviation of the gaussian kernel
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+        data_range: Range of the image. If ``None``, it is determined from the image (max - min)
+        k1: Parameter of SSIM.
+        k2: Parameter of SSIM.
+        betas: Exponent parameters for individual similarities and contrastive sensitivies returned by different image
+            resolutions.
+        normalize: When MultiScaleSSIM loss is used for training, it is desirable to use normalizes to improve the
+            training stability. This `normalize` argument is out of scope of the original implementation [1], and it is
+            adapted from https://github.com/jorge-pessoa/pytorch-msssim instead.
+
+    Return:
+        Tensor with Multi-Scale SSIM score
+
+    Example:
+        >>> from torchmetrics import MultiScaleSSIM
+        >>> preds = torch.rand([1, 1, 256, 256], generator=torch.manual_seed(42))
+        >>> target = preds * 0.75
+        >>> ssim = MultiScaleSSIM()
+        >>> ssim(preds, target)
+        tensor(0.9558)
+
+    References:
+    [1] Multi-Scale Structural Similarity For Image Quality Assessment by Zhou Wang, Eero P. Simoncelli and Alan C.
+    Bovik `MultiScaleSSIM`_
+    """
+
+    preds: List[Tensor]
+    target: List[Tensor]
+    higher_is_better = True
+    is_differentiable = True
+
+    def __init__(
+        self,
+        kernel_size: Sequence[int] = (11, 11),
+        sigma: Sequence[float] = (1.5, 1.5),
+        reduction: str = "elementwise_mean",
+        data_range: Optional[float] = None,
+        k1: float = 0.01,
+        k2: float = 0.03,
+        betas: Union[Tuple[float, float, float, float, float], Tuple[float, ...]] = (
```

Comment:
what is the diff between `Tuple[float, float, float, float, float]` and `Tuple[float, ...]`?

Reply:
A default setting considers `Tuple[float, float, float, float, float]`, however, in general, one can passes tuples of different sizes. Maybe `Tuple[float, ...]` is sufficient, right?